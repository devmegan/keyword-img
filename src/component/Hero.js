import Description from './Description'
import SearchInput from './SearchInput';

export default function Hero() {
    return (
        <div className="hero min-h-screen bg-base-200">
            <div className="hero-content text-center">
                <div className="max-w-md grid grid-cols-1 gap-3">
                    <span className="mx-auto inline-block">
                      <SearchInput />
                    </span>
                   <Description />
                </div>
            </div>
        </div>
    )
}