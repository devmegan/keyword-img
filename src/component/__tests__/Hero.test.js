import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Hero from '../Hero';

describe('description', () => {
    beforeEach(() => {
        jest.spyOn(console, 'error').mockImplementation(() => {});
    });

    it('renders w/o crashing', () => {
        expect.hasAssertions();

        const view = render(<Hero />);

        expect(view).toMatchSnapshot();
    });
});