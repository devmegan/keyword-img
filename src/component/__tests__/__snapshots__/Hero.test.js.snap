// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`description renders w/o crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="min-h-screen bg-base-100 w-full py-10 md:py-20"
      >
        <div
          class="max-w-screen-md px-10 md:px-0 grid grid-cols-1 gap-3 block mx-auto text-center"
        >
          <div
            class="h-80"
          />
          <span
            class="mx-auto inline-block"
          >
            <div
              class="input-group"
            >
              <input
                class="input input-bordered input-primary"
                placeholder="Keyword..."
                type="text"
                value=""
              />
              <button
                class="btn btn-square btn-primary"
              >
                <svg
                  class="h-6 w-6"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </button>
            </div>
          </span>
          <h1
            class="text-2xl font-bold"
          >
            Enter a keyword...
          </h1>
          <p>
            To receive an image associated to it
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="min-h-screen bg-base-100 w-full py-10 md:py-20"
    >
      <div
        class="max-w-screen-md px-10 md:px-0 grid grid-cols-1 gap-3 block mx-auto text-center"
      >
        <div
          class="h-80"
        />
        <span
          class="mx-auto inline-block"
        >
          <div
            class="input-group"
          >
            <input
              class="input input-bordered input-primary"
              placeholder="Keyword..."
              type="text"
              value=""
            />
            <button
              class="btn btn-square btn-primary"
            >
              <svg
                class="h-6 w-6"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </div>
        </span>
        <h1
          class="text-2xl font-bold"
        >
          Enter a keyword...
        </h1>
        <p>
          To receive an image associated to it
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
